packages/gestalt/src/HelpButton.js

```diff
@@ -0,0 +1,202 @@
+ */
+export default function HelpButton({
+  accessibilityPopoverLabel,
+  idealDirection = 'down',
+  link,
+  onClick,
+  text,
+  zIndex,
+}: Props): Node {
+  const tapAreaRef = useRef(null);
+  const [open, setOpen] = useState(false);
+  const [hovered, setHovered] = useState(false);
+  const [focused, setFocused] = useState(false);
+  const { name: colorSchemeName } = useColorScheme();
+  const popoverId = useId();
+  const { accessibilityTooltipMessage, accessibilityIconLabel } =
+    useDefaultLabelContext('HelpButton');
+
+  useEffect(() => {
+    if (open) {
+      tapAreaRef?.current?.focus();
+    }
+  }, [open]);
+
+  const toggleView = () => {
+    setOpen((currVal) => !currVal);
+  };
+
+  const onHandleTap = (props) => {
+    toggleView();
+    onClick?.(props);
+  };
+
+  const bgIconColor = open || hovered || focused ? 'default' : 'subtle';
+
+  const tooltipZIndex = zIndex ?? new FixedZIndex(1);
+
+  const zIndexWrapper = new CompositeZIndex([new FixedZIndex(tooltipZIndex.index())]);
+
+  // Overriding color of `Text` components
+  const isDarkMode = colorSchemeName === 'darkMode';
+  const textColorOverrideStyles = isDarkMode
+    ? styles.textColorOverrideLight
+    : styles.textColorOverrideDark;
+
+  const textElement =
+    typeof text === 'string' ? (
+      <Text align="start">{text}</Text>
+    ) : (
+      <span className={textColorOverrideStyles}>{text}</span>
+    );
+
+  return (
+    <Flex alignItems="center" justifyContent="center" flex="none">
+      <Tooltip
+        text={accessibilityTooltipMessage}
+        idealDirection={idealDirection}
+        zIndex={tooltipZIndex}
+      >
+        <TapArea
+          accessibilityExpanded={open}
+          accessibilityControls={popoverId}
+          accessibilityLabel={accessibilityTooltipMessage}
+          fullWidth={false}
+          onTap={onHandleTap}
+          ref={tapAreaRef}
+          role="button"
+          rounding="circle"
+          onBlur={() => setFocused(false)}
+          onFocus={() => setFocused(true)}
+          onMouseEnter={() => setHovered(true)}
+          onMouseLeave={() => setHovered(false)}
+        >
+          <Box
+            width={16}
+            height={16}
+            rounding="circle"
+            display="flex"
+            alignItems="center"
+            justifyContent="center"
+            dangerouslySetInlineStyle={{
+              __style: {
+                backgroundColor: `var(--color-text-${bgIconColor})`,
+              },
+            }}
+          >
+            <Icon
+              accessibilityLabel={accessibilityIconLabel}
+              color="inverse"
+              icon="question-mark"
+              size={8}
+            />
+          </Box>
+        </TapArea>
+      </Tooltip>
+      {open && (
+        <Box zIndex={zIndexWrapper}>
```

Comment:
Can you add a comment here that the only purpose of this Box is to make Zindex work (Tooltip over Popover) due not using Layer because by adding Layer you would have to bring focus to the Text element on opening the Popover, but instead you chose to keep the Text tabbable in the main sequential tabbing flow?

Reply:
Done